src/libbson/tests/test-bson-vector.c

```diff
@@ -0,0 +1,1581 @@
+   }
+   {
+      bson_iter_t a, b;
+      ASSERT (bson_iter_init_find (&a, &doc, "longer_vector"));
+      ASSERT (bson_iter_init_find (&b, &doc, "longer_round_trip"));
+      ASSERT (bson_iter_binary_equal (&a, &b));
+   }
+
+   // Padding bits will be initialized to zero when a packed_bit vector is first allocated by
+   // bson_append_vector_packed_bit_uninit
+   {
+      // Set the uninitialized part of 'doc' to a known value
+      static const uint32_t reserve_len = 512;
+      memset (bson_reserve_buffer (&doc, doc.len + reserve_len) + doc.len, 0xdd, reserve_len);
+
+      bson_vector_packed_bit_view_t view;
+      ASSERT (BSON_APPEND_VECTOR_PACKED_BIT_UNINIT (&doc, "padding_init_test", 12, &view));
+      ASSERT (bson_vector_packed_bit_view_length_bytes (view) == 2);
+      ASSERT (bson_vector_packed_bit_view_padding (view) == 4);
+
+      // BSON validity only requires the low 4 bits to be zero, but the entire last
+      // byte will be zeroed by our implementation.
+      uint8_t bytes[2];
+      ASSERT (bson_vector_packed_bit_view_read_packed (view, bytes, sizeof bytes, 0));
+      ASSERT_CMPUINT ((unsigned) bytes[0], ==, 0xdd);
+      ASSERT_CMPUINT ((unsigned) bytes[1], ==, 0x00);
+   }
+
+   // Padding bits can't be forcibly given nonzero values using bson_vector_packed_bit_view_write_packed
+   {
+      bson_vector_packed_bit_view_t view;
+      ASSERT (BSON_APPEND_VECTOR_PACKED_BIT_UNINIT (&doc, "padding_mask_test", 13, &view));
+      ASSERT (bson_vector_packed_bit_view_length_bytes (view) == 2);
+      ASSERT (bson_vector_packed_bit_view_padding (view) == 3);
+
+      uint8_t bytes[2] = {0xff, 0xff};
+      ASSERT (bson_vector_packed_bit_view_write_packed (view, bytes, sizeof bytes, 0));
+      ASSERT (bson_vector_packed_bit_view_read_packed (view, bytes, sizeof bytes, 0));
+      ASSERT_CMPUINT ((unsigned) bytes[0], ==, 0xff);
+      ASSERT_CMPUINT ((unsigned) bytes[1], ==, 0xf8);
+   }
+
+   bson_destroy (&doc);
+}
+
+// Note: The effective MAX_TESTED_VECTOR_LENGTH is limited to 2047 on Windows due to RAND_MAX==0x7fff
+#define MAX_TESTED_VECTOR_LENGTH 10000
+#define FUZZ_TEST_ITERS 5000
+
+static void
+test_bson_vector_view_api_fuzz_int8 (void)
+{
+   size_t current_length = 0;
+   bson_t vector_doc = BSON_INITIALIZER;
+   int8_t *expected_elements = bson_malloc (MAX_TESTED_VECTOR_LENGTH * sizeof *expected_elements);
+   int8_t *actual_elements = bson_malloc (MAX_TESTED_VECTOR_LENGTH * sizeof *actual_elements);
+   for (int fuzz_iter = 0; fuzz_iter < FUZZ_TEST_ITERS; fuzz_iter++) {
+      unsigned r = (unsigned) rand ();
+      unsigned r_operation = r & 0xFu;
+      size_t r_param = r >> 4;
+
+      if (current_length == 0 || r_operation == 15) {
+         // Resize and fill
+         size_t new_length = (size_t) r_param % MAX_TESTED_VECTOR_LENGTH;
+         bson_reinit (&vector_doc);
+         bson_vector_int8_view_t view;
+         BSON_ASSERT (BSON_APPEND_VECTOR_INT8_UNINIT (&vector_doc, "vector", new_length, &view));
```

Comment:
```suggestion
         ASSERT (BSON_APPEND_VECTOR_INT8_UNINIT (&vector_doc, "vector", new_length, &view));
```
Recommend using `ASSERT*` test assertion macros in test code (test-specific error messages + includes some [additional operations](https://github.com/mongodb/mongo-c-driver/blob/d6f975d175ec42487f2a1e19f55c87a98f996cc2/src/libmongoc/tests/TestSuite.h#L115) for test failure messages) rather than `BSON_ASSERT`.